lib/bb/event.py

```diff
@@ -1,142 +1,446 @@
+    fire_class_handlers(event, d)
+    if worker_fire:
+        worker_fire(event, d)
+    else:
+        # If messages have been queued up, clear the queue
+        global _uiready, ui_queue
+        if _uiready and ui_queue:
+            for queue_event in ui_queue:
+                fire_ui_handlers(queue_event, d)
+            ui_queue = []
+        fire_ui_handlers(event, d)
+
+def fire_from_worker(event, d):
+    fire_ui_handlers(event, d)
+
+noop = lambda _: None
+def register(name, handler, mask=None, filename=None, lineno=None, data=None):
+    """Register an Event handler"""
 
-NotHandled = 0
-Handled = 1
-handlers = []
+    if data is not None and data.getVar("BB_CURRENT_MC"):
+        mc = data.getVar("BB_CURRENT_MC")
+        name = '%s%s' % (mc.replace('-', '_'), name)
 
-def tmpHandler(event):
-    """Default handler for code events"""
-    return NotHandled
+    # already registered
+    if name in _handlers:
+        if data is not None:
+            bbhands_mc = (data.getVar("__BBHANDLERS_MC") or set())
+            bbhands_mc.add(name)
+            data.setVar("__BBHANDLERS_MC", bbhands_mc)
+        return AlreadyRegistered
 
-def defaultTmpHandler():
-    tmp = "def tmpHandler(e):\n\t\"\"\"heh\"\"\"\n\treturn 0"
-    comp = compile(tmp, "tmpHandler(e)", "exec")
-    return comp
+    if handler is not None:
+        # handle string containing python code
+        if isinstance(handler, str):
+            tmp = "def %s(e):\n%s" % (name, handler)
+            try:
+                code = bb.methodpool.compile_cache(tmp)
+                if not code:
+                    if filename is None:
+                        filename = "%s(e)" % name
+                    code = compile(tmp, filename, "exec", ast.PyCF_ONLY_AST)
+                    if lineno is not None:
+                        ast.increment_lineno(code, lineno-1)
+                    code = compile(code, filename, "exec")
+                    bb.methodpool.compile_cache_add(tmp, code)
+            except SyntaxError:
+                logger.error("Unable to register event handler '%s':\n%s", name,
+                             ''.join(traceback.format_exc(limit=0)))
+                _handlers[name] = noop
+                return
+            env = {}
+            bb.utils.better_exec(code, env)
+            func = bb.utils.better_eval(name, env)
+            _handlers[name] = func
+        else:
+            _handlers[name] = handler
 
-def fire(event):
-    """Fire off an Event"""
-    for h in handlers:
-        if type(h).__name__ == "code":
-            exec(h)
-            if tmpHandler(event) == Handled:
-                return Handled
+        if not mask or '*' in mask:
+            _catchall_handlers[name] = True
         else:
-            if h(event) == Handled:
-                return Handled
-    return NotHandled
+            for m in mask:
+                if _event_handler_map.get(m, None) is None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)